TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/SupportTasks/ApiTrnRequests/Index.cshtml.cs

```diff
@@ -20,12 +24,18 @@ public class Index(TrsDbContext dbContext) : PageModel
     [FromQuery]
     public ApiTrnRequestsSortByOption? SortBy { get; set; }
 
-    public Result[]? Results { get; set; }
+    [BindProperty(SupportsGet = true)]
+    [FromQuery(Name = "page")]
+    public int? PageNumber { get; set; }
+
+    public ResultPage<Result>? Results { get; set; }
+
+    public PaginationViewModel? Pagination { get; set; }
 
     public async Task<IActionResult> OnGetAsync()
     {
-        var sortDirection = SortDirection ?? SupportUi.SortDirection.Ascending;
-        var sortBy = SortBy ?? ApiTrnRequestsSortByOption.Name;
+        var sortDirection = SortDirection ??= SupportUi.SortDirection.Ascending;
```

Comment:
I broke these two properties previously; we rely on SortDirection being not `null` to have the tag helpers do the right thing.